tutorials/hardhat-tutorial.md

```diff
@@ -0,0 +1,425 @@
+
+![Successful verification using hardhat-etherscan plugin](/images/tutorials/hardhat/hardhat3.png)
+
+## Testing {: #testing }
+
+A robust smart contract development workflow is incomplete without a testing suite. Hardhat has a number of tools that make it easy to write and run tests. In this section, you'll learn the basics of testing your smart contracts and some more advanced techniques such as using fixtures. 
+
+Hardhat tests are typically written with Mocha and Chai. [Mocha](https://mochajs.org/){target=_blank} is a Javascript testing framework and [Chai](https://www.chaijs.com/){target=_blank} is a BDD/TDD Javascript assertion library. BDD / TDD stands for behavior and test driven development respectively. Effective BDD / TDD necessitates writing your tests *before* writing your smart contract code. The structure of this tutorial doesn't strictly follow these guidelines, but you may wish to adopt these principles in your development workflow. Hardhat previously recommended the Hardhat Waffle plugin but has since [advised migrating to the Hardhat Toolbox](https://hardhat.org/hardhat-runner/docs/guides/migrating-from-hardhat-waffle){target=_blank}.
+
+### Configuring the Test File {: #configuring-the-test-file }
+
+To get started, create a folder called `tests` and a file named `Dao.js`. Then copy and paste the contents below to set up the initial structure of your test file. Be sure to read the comments as they can clarify the purpose of each line. 
+
+``` javascript
+const { ethers } = require("hardhat");
+require("@nomicfoundation/hardhat-toolbox");
+
+// Import Chai to use its assertion functions here.
+const { expect } = require("chai");
+
+// Use `loadFixture` to share common setups (or fixtures) between tests.
+const { loadFixture } = require("@nomicfoundation/hardhat-network-helpers");
+
+// Indicate the collator the DAO wants to delegate to
+const targetCollator = "{{ networks.moonbase.staking.candidates.address1 }}";
+```
+
+### Using Fixtures {: #using-fixtures }
+
+A fixture is a function that configures the blockchain to a desired state. For example, our initial tests for the StakingDAO will be run after the StakingDAO has been deployed but before any contributions have been made. When a fixture is loaded for the first time, it is executed as normal to properly configure the state of the chain. When called a second time, the fixture is not executed again but rather the state of the network is reset to the point in time immediately after the fixture was initially executed. In other words, this resets state changes prior to the initial execution of the fixture and saves time.
```

Comment:
```suggestion
A fixture is a function that configures the blockchain to a desired state. For example, our initial tests for the staking DAO will be run after the staking DAO has been deployed but before any contributions have been made. When a fixture is loaded for the first time, it is executed as normal to properly configure the state of the chain. When called a second time, the fixture is not executed again but rather the state of the network is reset to the point in time immediately after the fixture was initially executed. In other words, this resets state changes prior to the initial execution of the fixture and saves time.
```

Reply:
fix all other occurrences... should be "`DelegationDAO`" (to match contract name) or "staking DAO"